romancal/flux/flux_step.py

```diff
@@ -48,33 +48,38 @@ class FluxStep(RomanStep):
 
     def process(self, input):
         if isinstance(input, datamodels.DataModel):
-            input_models = ModelContainer([input])
+            input_models = ModelLibrary([input])
             single_model = True
         elif isinstance(input, str):
             # either a single asdf filename or an association filename
             try:
                 # association filename
-                input_models = ModelContainer(input)
+                input_models = ModelLibrary(input)
                 single_model = False
             except Exception:
                 # single ASDF filename
-                input_models = ModelContainer([input])
+                input_models = ModelLibrary([datamodels.open(input)])
                 single_model = True
-        elif isinstance(input, ModelContainer):
+        elif isinstance(input, ModelLibrary):
             input_models = input
             single_model = False
         else:
             raise TypeError(
-                "Input must be an ASN filename, a ModelContainer, "
+                "Input must be an ASN filename, a ModelLibrary, "
                 "a single ASDF filename, or a single Roman DataModel."
             )
```

Comment:
Thanks! I think that's a good idea. I wonder if adding some of this logic to [RomanStep._datamodels_open](https://github.com/spacetelescope/romancal/blob/f593a51435bab3719849a248bf50e25db1ec5dc6/romancal/stpipe/core.py#L36) might help here (and other places). That classmethod is called by stpipe on the step input when attempting to resolve the config, reference files, etc. If it accepted:
- string -> if '.asdf' open as DataModel if '.json' (or '.yaml') open as ModelLibrary
- DataModel -> return DataModel
- ModelLibrary -> return ModelLibrary
then I think roman would be able to use `pars-` files, prefetch references, and any other stpipe infrastructure that uses `_datamodels_open`. I think we would still need some logic in this step (or in ModelLibrary) as we want a ModelLibrary even if we have a DataModel. It might boil down to something like:
```python
def process(input):
    opened_input = self._datamodels_open(input)
    if isinstance(opened_input, DataModel):
        input_models = ModelLibrary([opened_input])
    else:
        input_models = opened_input
```
One other issue to consider is how modification of the input is treated. With jwst there is a general "rule-of-thumb" that the input is not modified. With the above psuedocode if input is a model a copy would be need to be made to avoid modifying the input. Is there a similar convention for roman?

Reply:
> One other issue to consider is how modification of the input is treated. With jwst there is a general "rule-of-thumb" that the input is not modified. With the above psuedocode if input is a model a copy would be need to be made to avoid modifying the input. Is there a similar convention for roman?

I'm not 100% sure, but I don't think we have that for Roman.

Reply:
I'm going to mark this as "resolved". I agree that refactoring this and moving it to some common location is a great idea. However I'd like to restrict the scope of this PR (which is already quite large). Let me know if it's helpful to open an issue or ticket for this.

Reply:
I opened an issue to track this change: https://github.com/spacetelescope/romancal/issues/1309